hw_7/hw_7.2.js

```diff
@@ -0,0 +1,123 @@
+// Task 2.
+
+// subtask 1
+/*
+1. На вход функции подаётся предложение, например “I am the best AQA ever!” Преобразуйте строку таким образом, чтобы вместо каждой буквы была цифра, показывающая сколько раз эта буква встречается в предложении. 
+Пробелы и знаки препинания оставляем без изменения. Регистр не должен играть роли.
+*/
+const str = 'I am the best AQA ever!';
+
+function transformToAppearanceTimes(string) {
+	if (typeof string !== 'string') {
+		throw new Error('Something went wrong! Only string type for entered data is valid.');
+	} else {
+		const arr = str.toLowerCase().split('');
+
+		const transformedArray = arr.map((el, ind, arr) => {
+			if (el.charCodeAt(0) >= 97 && el.charCodeAt(0) <= 122) {
+				let counter = 0;
+
+				for (let i = 0; i < arr.length; i++) {
+					if (arr[ind] === arr[i]) {
+						counter++;
+					}
+				}
+				return counter;
+			} else return el;
+		})	
+
+		return transformedArray.join('');
+	}
+}
+console.log(transformToAppearanceTimes(str));
+
+
+// sabtask 2
+/*
+2. У вас есть массив с ценами товаров в чеке. В консоль нужно вывести сумму всех цен и среднюю цену товара.
+  Итого: 8495 $, средняя цена товара 700 $ - пример сообщения в консоле.  
+  const prices = [64, 7556, 345, 7556, 345, 7556, 345, 7556, 433, 345, 756, 123, 942, 3112, 421, 9341, 1212, 8, 43, 41, 345, 341, 21, 321, 123];
+*/
+
+const prices = [64, 7556, 345, 7556, 345, 7556, 345, 7556, 433, 345, 756, 123, 942, 3112, 421, 9341, 1212, 8, 43, 41, 345, 341, 21, 321, 123];
+
+function getPriceInfo(priceTotal, priceAverage) {
+	return `Total price of all items is ${priceTotal}$. Average price of item is ${priceAverage}$.`;
+}
+
+function getTotalPrice(arr) {
+	const sum = arr.reduce((total, value) => total + value, 0);
+	return sum;
+}
```

Comment:
Try to use `arrow` functions :)
```suggestion
const getTotalPrice = (arr) => arr.reduce((total, value) => total + value, 0);
```